src/Cabal2JSON.hs

```diff
@@ -60,12 +62,25 @@ import Distribution.Verbosity as Cabal
 import Language.Haskell.Extension
 import System.Environment
 import Text.Show.Pretty (pPrint)
+import Data.List.NonEmpty (fromList)
+import Distribution.Types.PkgconfigName
+import Distribution.Types.PkgconfigVersionRange
+import Distribution.Types.PkgconfigVersion
+import Text.PrettyPrint
+import Distribution.Parsec (Parsec(parsec), CabalParsing (parsecHaskellString), eitherParsec)
+import Distribution.Compat.Lens (_1)
+import Distribution.PackageDescription
+import qualified Distribution.SPDX as SPDX
+import Distribution.SPDX (LicenseExpression(..), SimpleLicenseExpression(..))
+import Data.Text.Encoding
+import Data.Text (pack)
 
 cabal2JSON :: IO ()
 cabal2JSON = do
-  arg : _ <- getArgs
-  cabalFile <- Cabal.readGenericPackageDescription Cabal.deafening arg
-  pPrint cabalFile
+  arg <- getArgs
+  case arg of
+    [] -> error "Please provide path to cabal file as CLI argument"
+    arg -> Cabal.readGenericPackageDescription Cabal.deafening (head arg) >>= pPrint
```

Comment:
@DavHau This says `pPrint`.
You can use this to encode the file:
`SB8.putStrLn . encodeJSONViaCodec`
with
`import qualified Data.ByteString.Char8 as SB8`